odl/tomo/backends/util.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2014-2020 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Utilities for tomo backends."""
+
+from functools import wraps
+
+__all__ = ()
+
+
+def _add_default_complex_impl(fn):
+    """Wrapper to call a real-valued class method twice when input is complex.
+
+    This function helps `RayTransform` implementations with extending methods
+    that work for real-valued elements to complex-valued elements, by splitting
+    complex calls into two individual real calls.
+
+    The wrapper will only work for methods of which the class provides a
+    `vol_space` and `proj_space`. Typically, this will then work as a decorator
+    on the method, e.g.:
+    ```
+        @_add_default_complex_impl
+        def call_forward(self, x, out=None, **kwargs):
+            # Code that should run for real input and output
+    ```
```

Comment:
```suggestion
    on the method, e.g. ::

        @_add_default_complex_impl
        def call_forward(self, x, out=None, **kwargs):
            # Code that should run for real input and output
```

Note that the docstrings are parsed as reStructuredText, not Markdown. So the triple-quote thing won't work.